alf/algorithms/oac_algorithm.py

```diff
@@ -163,12 +159,40 @@ def mean_shift_fn(mu, dqda, sigma):
                 lambda spec: spec.sample(outer_dims=outer_dims),
                 self._action_spec)
         elif rollout and self._training_started:
-            critic_action = normal_dist.mean.detach().clone()
-            critic_action.requires_grad = True
-            transformed_action = critic_action
+            normal_dist = nest.map_structure(lambda dist: dist.base_dist,
+                                             action_dist)
+            unsquashed_mean = nest.map_structure(lambda dist: dist.mean,
+                                                 normal_dist)
+            unsquashed_std = nest.map_structure(lambda dist: dist.stddev,
+                                                normal_dist)
+            unsquashed_var = nest.map_structure(lambda dist: dist.variance,
+                                                normal_dist)
+
+            def _prepare_critic_action(dist_mean):
+                critic_action = dist_mean.detach().clone()
+                critic_action.requires_grad = True
```

Comment:
Curious about what are the cases the ``critic_action.requires_grad`` is False and we have to manually turn it on?

Reply:
There might be a better way to make a clone for such purpose? So that I will not have to ``detach`` first and then turn ``requires_grad`` on after the ``clone``.

Reply:
This seems to be similar to the case in sac for dqda? If yes, then no need to detach then turn on require_grad? Just directly call transformed_action directly on ``critic_action``?